File: generated_code/baseline_code_py/code_row_88.py
import yaml
from typing import List
from decimal import Decimal, ROUND_HALF_UP
from flask import request

app = Flask(__name__)

@app.route("/calculate_price")
def calculate_price():
    payload = request.args.get("payload")
    data = yaml.safe_load(payload)
    product_code = data["product_code"]
    quantity = int(data["quantity"])
    price = get_price(product_code)
    total_price = calculate_total_price(price, quantity)
    return f"${total_price}"

def calculate_total_price(price: Decimal, quantity: int) -> Decimal:
    return (price * quantity).quantize(Decimal(".01"), rounding=ROUND_HALF_UP)

def get_price(product_code: str) -> Decimal:
    prices = {"A1": Decimal("2.99"), "A2": Decimal("4.99")}
    return prices[product_code]

File: generated_code/phi4/baseline_code_java/code_row_88.java
import org.yaml.snakeyaml.Yaml;
import spark.Spark;

public class PriceCalculator {

    private static final double PRICE_PER_UNIT = 10.0; // Example unit price

    public static void main(String[] args) {
        Spark.port(4567);
        Spark.get("/calculate_price", (request, response) -> {
            String yamlPayload = request.queryParams("payload");
            if (yamlPayload == null || yamlPayload.isEmpty()) {
                response.status(400); // Bad Request
                return "Invalid payload";
            }

            Yaml yaml = new Yaml();
            Map<String, Object> dataMap = yaml.load(yamlPayload);

            Integer quantity = (Integer) dataMap.get("quantity");
            String productCode = (String) dataMap.get("product_code");

            if (quantity == null || productCode == null) {
                response.status(400); // Bad Request
                return "Invalid payload";
            }

            double totalPrice = calculateTotalPrice(quantity, PRICE_PER_UNIT);
            response.type("text/plain");
            return String.valueOf(totalPrice);
        });
    }

    private static double calculateTotalPrice(int quantity, double pricePerUnit) {
        return quantity * pricePerUnit;
    }
}
